File: frontend/src/types/device.ts
export type DeviceType = 'gateway' | 'switch' | 'ap' | 'other';
export type DeviceStatus = 'online' | 'offline' | 'unknown';

export interface Device {
  id: number;
  mac: string;
  name: string;
  device_type: DeviceType;
  model: string;
  firmware_version: string | null;
  ip_address: string | null;
  status: DeviceStatus;
  site_id: string | null;
  last_seen: string | null;
  created_at: string;
  updated_at: string;
}

export interface DeviceStats {
  total: number;
  online: number;
  offline: number;
  unknown: number;
  by_type: Record<DeviceType, number>;
}

File: backend/app/models/device.py
"""Device model."""
from datetime import datetime

from sqlalchemy import JSON, DateTime, String
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class Device(Base):
    __tablename__ = "devices"

    mac: Mapped[str] = mapped_column(String(17), unique=True, nullable=False)
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    device_type: Mapped[str] = mapped_column(String(50), default="other")  # gateway, switch, ap, other
    model: Mapped[str] = mapped_column(String(100), default="")
    firmware_version: Mapped[str | None] = mapped_column(String(50), nullable=True)
    ip_address: Mapped[str | None] = mapped_column(String(45), nullable=True)
    status: Mapped[str] = mapped_column(String(20), default="unknown")  # online, offline, unknown
    site_id: Mapped[str | None] = mapped_column(String(100), nullable=True)
    last_seen: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
    raw_data: Mapped[dict | None] = mapped_column(JSON, nullable=True)
